bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlBaseHandler.java

```diff
@@ -0,0 +1,140 @@
+            logger.debug("communication not up yet, cannot handle command {} for {}", command, channelUID);
+            return;
+        }
+
+        // This can be expensive, therefore do it in a job.
+        scheduler.submit(() -> {
+            if (!nhcComm.communicationActive()) {
+                restartCommunication(nhcComm);
+            }
+
+            if (nhcComm.communicationActive()) {
+                handleCommandSelection(channelUID, command);
+            }
+        });
+    }
+
+    abstract void handleCommandSelection(ChannelUID channelUID, Command command);
+
+    @Override
+    public void deviceInitialized() {
+        logger.debug("device initialized {}", deviceId);
+
+        Bridge bridge = getBridge();
+        if ((bridge != null) && (bridge.getStatus() == ThingStatus.ONLINE)) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+
+        refresh();
+    }
+
+    abstract void refresh();
+
+    @Override
+    public void deviceRemoved() {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                "@text/offline.configuration-error.deviceRemoved");
+    }
+
+    abstract void startCommunication();
+
+    void restartCommunication(NikoHomeControlCommunication nhcComm) {
+        // We lost connection but the connection object is there, so was correctly started.
+        // Try to restart communication.
+        nhcComm.scheduleRestartCommunication();
+        // If still not active, take thing offline and return.
+        if (!nhcComm.communicationActive()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error");
+            return;
+        }
+        // Also put the bridge back online
+        NikoHomeControlBridgeHandler nhcBridgeHandler = getBridgeHandler();
+        if (nhcBridgeHandler != null) {
+            nhcBridgeHandler.bridgeOnline();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                    "@text/offline.bridge-unitialized");
+        }
+    }
+
+    @Nullable
+    NikoHomeControlCommunication getCommunication(@Nullable NikoHomeControlBridgeHandler nhcBridgeHandler) {
+        return nhcBridgeHandler != null ? nhcBridgeHandler.getCommunication() : null;
+    }
+
+    @Nullable
+    NikoHomeControlBridgeHandler getBridgeHandler() {
+        Bridge nhcBridge = getBridge();
+        return nhcBridge != null ? (NikoHomeControlBridgeHandler) nhcBridge.getHandler() : null;
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        ThingStatus bridgeStatus = bridgeStatusInfo.getStatus();
+        if (ThingStatus.ONLINE.equals(bridgeStatus)) {
+            if (!initialized) {
+                scheduler.submit(this::startCommunication);
```

Comment:
This thread is not monitored.
I imagine it could be required to "cancel" it in case the thing handler is disposed.

Reply:
Yes, indeed, I guess this could happen when disposing during the start of the communication. I will monitor and cancel in dispose.